pkg/cc_dist_library.bzl

```diff
@@ -265,6 +272,27 @@ def _collect_inputs(deps):
         ),
     )
 
+# Given structs a and b returned from _collect_inputs(), returns a copy of a
+# but with all files from b subtracted out.
+def _subtract_files(a, b):
+    result_args = {}
+
+    top_level_fields = ["objects", "pic_objects"]
+    for field in top_level_fields:
+        to_remove = {e: None for e in getattr(b, field)}
+        result_args[field] = [e for e in getattr(a, field) if not e in to_remove]
+
+    cc_file_list_args = {}
+    file_list_fields = ["srcs", "hdrs", "internal_hdrs", "textual_hdrs"]
+    for field in file_list_fields:
+        to_remove = {e: None for e in getattr(b.cc_file_list, field).to_list()}
+        cc_file_list_args[field] = depset(
+            [e for e in getattr(a.cc_file_list, field).to_list() if not e in to_remove and e.path.find("external") == -1],
```

Comment:
Clever, is this how we're filtering out non-proto dependencies?  It might be worth a comment given the importance here

Reply:
Right, it's filtering out non-proto dependencies by checking for "external" in the file name. (There might actually be a better way to do it, but this is the best I've figured out so far.) I added a comment to mention this.

Reply:
The way I thought you were going to do it after our discussion a while back was to force the cc_dist_library to explicitly exclude these in some way, so this is a huge improvement over that :)

I think one way to handle this a little better is in `_flatten_target_files`, you can use `target.label.workspace_name` to find the repo that the target is in.  I don't think you want to search past external deps, so it's better to stop the transitive search once you hit them anyway

Reply:
See here: https://bazel.build/rules/lib/Target, https://bazel.build/rules/lib/Label

Reply:
Oh, good idea! Done.